homeassistant/components/climacell/__init__.py

```diff
@@ -1,96 +1,14 @@
-from pyclimacell.exceptions import (
-    CantConnectException,
-    InvalidAPIKeyException,
-    RateLimitedException,
-    UnknownException,
-)
-
 from homeassistant.components.tomorrowio import DOMAIN as TOMORROW_DOMAIN
 from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
-from homeassistant.const import (
-    CONF_API_KEY,
-    CONF_API_VERSION,
-    CONF_LATITUDE,
-    CONF_LONGITUDE,
-    CONF_NAME,
-    Platform,
-)
+from homeassistant.const import CONF_API_VERSION
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-from homeassistant.helpers.device_registry import DeviceEntryType
-from homeassistant.helpers.entity import DeviceInfo
-from homeassistant.helpers.update_coordinator import (
-    CoordinatorEntity,
-    DataUpdateCoordinator,
-    UpdateFailed,
-)
-
-from .const import (
-    ATTRIBUTION,
-    CC_V3_ATTR_CLOUD_COVER,
-    CC_V3_ATTR_CONDITION,
-    CC_V3_ATTR_HUMIDITY,
-    CC_V3_ATTR_OZONE,
-    CC_V3_ATTR_PRECIPITATION,
-    CC_V3_ATTR_PRECIPITATION_DAILY,
-    CC_V3_ATTR_PRECIPITATION_PROBABILITY,
-    CC_V3_ATTR_PRECIPITATION_TYPE,
-    CC_V3_ATTR_PRESSURE,
-    CC_V3_ATTR_TEMPERATURE,
-    CC_V3_ATTR_VISIBILITY,
-    CC_V3_ATTR_WIND_DIRECTION,
-    CC_V3_ATTR_WIND_GUST,
-    CC_V3_ATTR_WIND_SPEED,
-    CC_V3_SENSOR_TYPES,
-    CONF_TIMESTEP,
-    DEFAULT_TIMESTEP,
-    DOMAIN,
-    MAX_REQUESTS_PER_DAY,
-)
-
-_LOGGER = logging.getLogger(__name__)
 
-PLATFORMS = [Platform.SENSOR, Platform.WEATHER]
-
-
-def _set_update_interval(hass: HomeAssistant, current_entry: ConfigEntry) -> timedelta:
-    """Recalculate update_interval based on existing ClimaCell instances and update them."""
-    api_calls = 4 if current_entry.data[CONF_API_VERSION] == 3 else 2
-    # We check how many ClimaCell configured instances are using the same API key and
-    # calculate interval to not exceed allowed numbers of requests. Divide 90% of
-    # MAX_REQUESTS_PER_DAY by 4 because every update requires four API calls and we want
-    # a buffer in the number of API calls left at the end of the day.
-    other_instance_entry_ids = [
-        entry.entry_id
-        for entry in hass.config_entries.async_entries(DOMAIN)
-        if entry.entry_id != current_entry.entry_id
-        and entry.data[CONF_API_KEY] == current_entry.data[CONF_API_KEY]
-    ]
-
-    interval = timedelta(
-        minutes=(
-            ceil(
-                (24 * 60 * (len(other_instance_entry_ids) + 1) * api_calls)
-                / (MAX_REQUESTS_PER_DAY * 0.9)
-            )
-        )
-    )
-
-    for entry_id in other_instance_entry_ids:
-        if entry_id in hass.data[DOMAIN]:
-            hass.data[DOMAIN][entry_id].update_interval = interval
-
-    return interval
+from .const import CONF_TIMESTEP, DEFAULT_TIMESTEP, DOMAIN
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
How long should we keep converting / importing this?

Reply:
good question, I am not sure what the right answer is. Climacell has been removed from our docs since 2022.4 and the v3 API (the one that the integration still supports) is no longer available. So this would only help users who are updating from a pre 2022.4 setup to anything after the next release.

I also wasn't able to find this integration in analytics.

Reply:
![CleanShot 2022-09-22 at 19 11 12](https://user-images.githubusercontent.com/195327/191809875-ab26437e-cca5-4f57-ab08-8474c7769127.png)

35

Reply:
huh do we cap the UI list? Can't find it on the analytics site

Reply:
Yes, the list is capped by the integration being listed on the docs.